packages/zendesk-support-adapter/src/filters/custom_field_options/creator.ts

```diff
@@ -0,0 +1,188 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import {
+  Change, getChangeData, InstanceElement, isAdditionChange, isInstanceElement,
+  Values, Element, isObjectType, Field, BuiltinTypes, ReferenceExpression,
+} from '@salto-io/adapter-api'
+import { deployment, client as clientUtils } from '@salto-io/adapter-components'
+import { collections } from '@salto-io/lowerdash'
+import { getParents } from '@salto-io/adapter-utils'
+import { FilterCreator } from '../../filter'
+import { addIdUponAddition, deployChange, deployChanges } from '../../deployment'
+import { applyforInstanceChangesOfType } from '../utils'
+
+export const CUSTOM_FIELD_OPTIONS_FIELD_NAME = 'custom_field_options'
+export const DEFAULT_CUSTOM_FIELD_OPTION_FIELD_NAME = 'default_custom_field_option'
+
+type CustomFieldOptionsFilterCreatorParams = {
+  parentTypeName: string
+  childTypeName: string
+}
+
+const { makeArray } = collections.array
+
+export const createCustomFieldOptionsFilterCreator = (
+  { parentTypeName, childTypeName }: CustomFieldOptionsFilterCreatorParams
+): FilterCreator => ({ config, client }) => ({
+  onFetch: async (elements: Element[]): Promise<void> => {
+    const parentType = elements
+      .filter(isObjectType)
+      .find(inst => inst.elemID.typeName === parentTypeName)
+    if (parentType === undefined) {
+      return
+    }
+    parentType.fields[DEFAULT_CUSTOM_FIELD_OPTION_FIELD_NAME] = new Field(
+      parentType,
+      DEFAULT_CUSTOM_FIELD_OPTION_FIELD_NAME,
+      BuiltinTypes.STRING,
+    )
+    const parentInstances = elements
+      .filter(isInstanceElement)
+      .filter(inst => inst.elemID.typeName === parentTypeName)
+      .filter(inst => inst.value[CUSTOM_FIELD_OPTIONS_FIELD_NAME] !== undefined)
+    parentInstances.forEach(inst => {
+      const options = elements
+        .filter(isInstanceElement)
+        .filter(childInst => childInst.elemID.typeName === childTypeName)
+        .filter(childInst => {
+          const parent = getParents(childInst)?.[0]
+          return parent !== undefined && (parent.value.value.id === inst.value.id)
+        })
+      const defaultOption = options.find(option => option?.value?.default === true)
+      if (defaultOption) {
+        inst.value[DEFAULT_CUSTOM_FIELD_OPTION_FIELD_NAME] = new ReferenceExpression(
+          defaultOption.elemID
```

Comment:
I think reference from fetch should be resolved  so you probably should 
```suggestion
          defaultOption.elemID,
          defaultOption,
          
```